docs/deploying.adoc

```diff
@@ -9,11 +9,250 @@ Put the filter jars in some directory
+interdependency; it is supported to have TLS configured for the downstream and use plain communications for the
+upstream, or vice-versa.
+
+NOTE: TLS is recommended for both upstream and downstream for production configurations.
+
+=== Downstream TLS
+
+Here's how to enable TLS for the downstream side. This means the Kafka Client will connect to the virtual cluster over
+TLS rather than plain.  For this, you will need to obtain a TLS certificate for the virtual cluster from your
+Certificate Authority.
+
+NOTE: When requesting the certificate ensure that the certificate will match the names of the virtual cluster's
+bootstrap and broker addresses.  This may mean making use of wildcard certificates and/or Subject Alternative Names (SANs).
+
+Kroxylicious accepts key material in PKCS12 or JKS keystore format, or PEM formatted file(s).  The following configuration
+illustrates configuration with PKCS12 keystore.
+
+[source, yaml]
+----
+virtualClusters:
+  demo:
+    tls:
+        key:
+          storeFile: /opt/cert/server.p12               # <1>
+          storePassword:
+            passwordFile: /opt/cert/store.password      # <2>
+          keyPassword:
+            passwordFile: /opt/cert/key.password        # <3>
+          storeType: PKCS12                             # <4>
+    clusterNetworkAddressConfigProvider:
+      ...
+----
+<1> File system location of a keystore (or in the case of `PEM` format a text file containing the concatenation of the
+private key, certificate, and intermediates).
+<2> File system location of a file containing the key store's password.
+<3> (Optional) File system location of a file containing the key's password. If omitted the key store's password is
+used to decrypt the key too.
+<4> (Optional) Store type. Supported types are: `PKCS12`, `JKS` and `PEM`.  Defaults to Java default key store type
+which is usually `PKCS12`.
+
+Alternatively, if your key material is in separate PEM files (private key, and certificate/intermediates), the following
+configuration may be used:
+
+[source, yaml]
+----
+virtualClusters:
+  demo:
+    tls:
+        key:
+          privateKeyFile: /opt/cert/server.key          # <1>
+          certificateFile: /opt/cert/server.crt         # <2>
+          keyPassword:
+            passwordFile: /opt/cert/key.password        # <3>
+    clusterNetworkAddressConfigProvider:
+      ...
+----
+<1> File system location of the server private key.
+<2> File system location of the server certificate and intermediate(s).
+<3> (Optional) File system location of a file containing the key's password.
+
+=== Upstream TLS
+
+Here's how to enable TLS for the upstream side.  This means that Kroxylicious connects to the (physical) Kafka Cluster)
+over TLS.  For this, your Kafka Cluster must have already been configured to use TLS.
+
+By default, Kroxylicious uses the trust of the platform to decide whether the Kafka Cluster is trusted or not.  To
```

Comment:
I find the 
> the trust of the platform 

to be quite an odd phrase. Personally I might go for 
`By default Kroxylicious inherets what it trusts from the platofrm its running on and uses this to determine whether the Kafka Cluster is trusted or not.`

Reply:
Fixed.